libs/AssistedJsonRpcProvider.js

```diff
@@ -147,7 +147,7 @@ class AssistedJsonRpcProvider extends Provider {
                     filter?.topics?.[1]?.[i],
                     filter?.topics?.[2]?.[i],
                     filter?.topics?.[3]?.[i],
-                ]
+                ].filter((topic) => topic !== undefined)
```

Comment:
try `!= null`

Reply:
with topics = [null, accTopic, undefined, undefined] 
`!= null` will remove topic0 => wrong

Reply:
then it should be `trimTrailingNullItems`